### DISCUSSION TITLE
[Noob developer] Can you please tell me if I can add next js to my Parcel project

### DISCUSSION THREAD

**Discussion by GedeonWest** [vercel/next.js#36673] [Help]
There is an existing project that uses Parcel to build. It is necessary to implement Next, but I don't understand how to implement it in start/dev/build.
Is there such a possibility?

**Comment by icyJoseph** [comment]
The next.js routing system makes it so that to progressively adapt next.js, you must use rewrites, and run both your old app and the next.js app at the same time. So that when a user requests a route you haven't migrated yet, you can, under the same URL, show the parcel App.

I'd say, it might be better and easier to migrate the entire thing to next.js in one go, if possible.

I've migrated a few apps from CRA to next.js, and after some discussions in our teams, we went with the all in one go approach.

https://nextjs.org/docs/migrating/incremental-adoption

The docs also talk about sub paths, in which your next.js app would be a sort of child to the parcel app, but I'd say rewrites are a bit better.

It must be said that sharing React state between these becomes a bit more challenging.
